lib/tasks/globus.rake

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+namespace :globus do
+  desc 'Delete Globus access rules and empty directories'
+  task :prune_access, [:sunet_id] => :environment do |_task, args|
+    globus_destinations = GlobusDestination.joins(:user).where('user.sunet_id': args[:sunet_id])
+    puts "No GlobusDestinations found for #{args[:sunet_id]}" if globus_destinations.empty?
+    globus_destinations.each do |dest|
+      if dest.deleted_at.nil?
+        begin
+          GlobusClient.delete_access_rule(path: dest.destination_path, user_id: "#{args[:sunet_id]}@stanford.edu")
+          puts "Deleted access to globus directory #{dest.destination_path} for #{args[:sunet_id]}"
+        rescue StandardError => e # catch any "Access rule not found" errors from Globus
+          puts e
+        end
+        dest.deleted_at = Time.now.utc
```

Comment:
Should marking the GlobusDestination as deleted be moved inside the `begin` block so that we don't mark things as deleted where we failed to talk to Globus?

Reply:
That makes sense. I guess it would be unusual to have a rule not (any longer) existing in this situation. We would want to investigate rather than updating the GlobusDestination. Fixed.